### ISSUE TITLE
wallet.pay() returns a 400 - "attachment.value must be of type object"

### ISSUE THREAD

**Issue by mdrabic** [HandCash/handcash-connect-sdk-js#26]
[Attachment](https://github.com/HandCash/handcash-connect-sdk-js/blob/master/types/index.d.ts#L137)'s property `value` is typed as a string. When making a call to `wallet.pay()` with payment parameters that include an attachment that adheres to the typing, the follow error is thrown: 

```json
{"httpStatusCode":400,"message":"\"attachment.value\" must be of type object","info":{"path":["attachment","value"],"type":"object.base","context":{"type":"object","label":"attachment.value","value":"{\"app\":\"Take It\",\"action\":\"Paid Vote\",\"collection\":\"A  sample collection\",\"voteCount\":\"1\"}","key":"value"}}}
```

Example that produces this 400 response

```js
      const paymentParameters = {
        description: 'Nft Drop Vote',
        payments: [{ destination: takeItHandle, currencyCode: 'USD' as CurrencyCode, sendAmount: 1.0, tags: tag }],
        attachment: {
          format: 'json',
          value: JSON.stringify({ app: 'Take It', action: 'Paid Vote', collection: collection.name, voteCount: '1' }),
        },
      } as PaymentParameters;

     await account.wallet.pay(paymentParameters);
```

**Comment by rseibane** [comment]
Hi @mdrabic, thanks for reporting this issue.

As I understand the error in the context, the validator is expecting `attachment.value` to be of type `object` but you are sending it as a serialized `string`. Could you try with the following syntax and let me know?
```
const paymentParameters = {
      description: 'Nft Drop Vote',
      payments: [{ destination: takeItHandle, currencyCode: 'USD' as CurrencyCode, sendAmount: 1.0, tags: tag }],
      attachment: {
        format: 'json',
        value: { app: 'Take It', action: 'Paid Vote', collection: collection.name, voteCount: '1' },
      },
} as PaymentParameters;

await account.wallet.pay(paymentParameters);
```

**Comment by mdrabic** [comment]
@rseibane Our project is typescript based and we have it setup to perform compile-time type checking. The above code block fails compilation because Attachment `value` is expected to be a string but an object is being set on it.

**Comment by rseibane** [comment]
This should be solved with #28. Let me know if now works as expected for you @mdrabic

**Comment by mdrabic** [comment]
I pulled 0.4.3 and the change is good. thanks!
